Src/AutoFixtureUnitTest/MailAddressGeneratorTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Net.Mail;
+using System.Text.RegularExpressions;
+using Ploeh.AutoFixture;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.AutoFixtureUnitTest.Kernel;
+using Xunit;
+
+namespace Ploeh.AutoFixtureUnitTest
+{
+    public class MailAddressGeneratorTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            // Fixture setup
+            // Exercise system
+            var sut = new MailAddressGenerator();
+            // Verify outcome
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNullRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var sut = new MailAddressGenerator();
+            // Exercise system
+            var dummyContext = new DelegatingSpecimenContext();
+            var result = sut.Create(null, dummyContext);
+            // Verify outcome
+            Assert.Equal(new NoSpecimen(), result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNonMailAddressRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var request = new object();
+            var sut = new MailAddressGenerator();
+            // Exercise system
+            var dummyContext = new DelegatingSpecimenContext();

```

Comment:
Is `dummyContext` part of the [Exercise SUT](http://xunitpatterns.com/exercise%20SUT.html) phase?